staging/src/k8s.io/client-go/transport/websocket/roundtripper.go

```diff
@@ -111,6 +113,14 @@ func (rt *RoundTripper) RoundTrip(request *http.Request) (retResp *http.Response
 	wsConn, resp, err := dialer.DialContext(request.Context(), request.URL.String(), request.Header)
 	if err != nil {
 		if errors.Is(err, gwebsocket.ErrBadHandshake) {
+			// Enhance the error message with the http.Response data if possible.
+			if resp != nil {
+				errInfo := bytes.NewBufferString(resp.Status)
+				if errorBody, readErr := io.ReadAll(resp.Body); readErr == nil && len(errorBody) != 0 {
+					errInfo.Write(errorBody)
+				}
+				err = fmt.Errorf("%w\n%s", err, errInfo.String())
```

Comment:
I'm not sure about splitting this across lines or joining with something like `: ` on the same line

Reply:
Removed.